pkg/server/nodemanager/node_controller.go

```diff
@@ -267,13 +270,45 @@ func (c *NodeController) startSingleNode(node *api.Node, image cloud.Image, clou
 	var (
 		instanceID string
 		err        error
+		excludeInstances []string
 	)
-	if node.Spec.Spot {
-		instanceID, err = c.CloudClient.StartSpotNode(node, image, cloudInitData, c.Config.DefaultIAMPermissions)
-	} else if node.Spec.Dedicated {
-		instanceID, err = c.CloudClient.StartDedicatedNode(node, image, cloudInitData, c.Config.DefaultIAMPermissions)
-	} else {
-		instanceID, err = c.CloudClient.StartNode(node, image, cloudInitData, c.Config.DefaultIAMPermissions)
+	for retries := 0; retries < NoCapacityRetryLimit; retries++ {
```

Comment:
What's the purpose of retrying here? That will be taken care of by the control loop.